proposals/0230-spp-resource-management-for-android.md

```diff
@@ -36,175 +35,139 @@ and define the setter:
+This is why Code-1 above uses different channel ID (SDL_ERROR_NOTIFICATION_CHANNEL_ID), and different setting of channel importance flag (IMPORTANCE_HIGH)
+
+When user opens "manage notifications", the app's notification setting will be shown below:
+
+![manage notification setting](../assets/proposals/0230-spp-resource-management-for-android/manage_notification_1.png)
+
+**Fig. 2: Manage notifications setting**
+
+Please note that channel name ("SDL Error" in Fig. 2) is TBD.
+
+And the channel ("SDL Error") should be configured like below:
+
+![manage notification category setting](../assets/proposals/0230-spp-resource-management-for-android/manage_notification_2.png)
+
+
+**Fig. 3: Manage notification category setting**
+
+### When user clicks on notification popup
+
+In previous code snippet (Code-1), SdlNotificationActivity is specified for ContentIntent. The activity is shown when user clicks on the notification popup.
+
+The sample UX of SdlNotificationActivity looks like below:
+
+![SdlNotificationActivity UX](../assets/proposals/0230-spp-resource-management-for-android/spp_error_activity.png)
+
+All strings used for this error UX is defined in strings.xml like below:
 
-    /**
-     * getLegacyModeAllowed (getter)
-     * @return
-     */
-    public boolean getLegacyModeAllowed() {
-        return this.legacyModeAllowed;
-    }
-}
-```
-```enterLegacyMode``` checks to see ```transportListener.onLegacyModeEnabled```, and we can check the flag in ```onLegacyModeEnabled``` something like below:
 ```java
-in SdlProtocol:
-
-        public boolean onLegacyModeEnabled(String info) {
-            //Await a connection from the legacy transport
-            if(transportConfig.getLegacyModeAllowed() && requestedPrimaryTransports!= null && requestedPrimaryTransports.contains(TransportType.BLUETOOTH)
-                    && !transportConfig.requiresHighBandwidth()){
-                Log.d(TAG, "Entering legacy mode; creating new protocol instance");
-                reset();
-                return true;
-            }else{
-                Log.d(TAG, "legacy mode is not allowed");
-                return false;
-            }
-        }
+    <string name="spp_out_of_resource">Bluetooth channel is out of resource</string>
+    <string name="notification_title">SmartDeviceLink</string>
+    <string name="spp_out_of_resource_message">There are too many bluetooth apps running on your device. Please close them and try to re-connect</string>
+    <string name="button_ok">OK</string>
+    <string name="sdl_error_notification_channel_name">SDL Error</string>
 ```
 
+The strings above are TBD (requires review).
+
+### Localization of SPP error UX
+
+Because all strings used for error UX is defined in strings.xml, the standard localization process for Android can be applied; i.e. adding language specific strings.xml per language.
```

Comment:
```suggestion
Because all strings used for error UX are defined in strings.xml, the standard localization process for Android can be applied; i.e. adding language specific strings.xml per language.
```

Reply:
Thanks for the suggestion. Updated accordingly.